tests/test_processor.py

```diff
@@ -222,3 +246,19 @@ def test_add_remove_ports():
     for i in range(6):
         assert processor.get_input_port(i) is None
         assert processor.get_output_port(i) is None
+
+
+def test_phase_quantization():
```

Comment:
This is an indirect test, could we also test the phase values of p2 in this test ?

Reply:
I only want to test that _manually reducing the precision of phase values_ has exactly the same effect as _using the `phase_imprecision` noise parameter_. I do not care about the exact probability results in this test - as this would be more like testing the simulation back-end than the noise handling of the `Processor` (so I really think this does not belong to this test).